aspnetcore/security/authentication/social/additional-claims.md

```diff
@@ -105,6 +108,122 @@ Users can define custom actions by deriving from <xref:Microsoft.AspNetCore.Auth
 
 For more information, see <xref:Microsoft.AspNetCore.Authentication.OAuth.Claims>.
 
+## Add and update user claims
+
+Claims are copied from external providers to the user database on first registration, not on sign in. If additional claims are enabled in an app after a user registers to use the app, call [SignInManager.RefreshSignInAsync](xref:Microsoft.AspNetCore.Identity.SignInManager%601) on a user to force the generation of a new authentication cookie.
+
+In the Development environment working with test user accounts, you can simply delete and recreate the user account. For production systems, new claims added to the app can be backfilled into user accounts. After [scaffolding the `ExternalLogin` page](xref:security/authentication/scaffold-identity) into the app at `Areas/Pages/Identity/Account/Manage`, add the following code to the `ExternalLoginModel` in the `ExternalLogin.cshtml.cs` file.
+
+Add a dictionary of added claims. Use the dictionary keys to hold the claim types, and use the values to hold a default value. Add the following line to the top of the class. The following example assumes that one claim is added for the user's Google picture with a generic headshot image as the default value:
+
+```csharp
+private readonly IDictionary<string, string> _addedClaims = 
+    new Dictionary<string, string>()
+    {
+        { "urn:google:picture", "https://localhost:5001/headshot.png" },
+    };
+```
+
+Replace the default code of the `OnGetCallbackAsync` method with the following code. The code loops through the claims dictionary. Claims are added (backfilled) or updated for each user. When claims are added or updated, the user sign-in is refreshed using the <xref:Microsoft.AspNetCore.Identity.SignInManager%601>, preserving the existing authentication properties (`AuthenticationProperties`).
+
+```csharp
+public async Task<IActionResult> OnGetCallbackAsync(
+    string returnUrl = null, string remoteError = null)
+{
+    returnUrl = returnUrl ?? Url.Content("~/");
+    if (remoteError != null)
+    {
+        ErrorMessage = $"Error from external provider: {remoteError}";
+        return RedirectToPage("./Login", new {ReturnUrl = returnUrl });
+    }
+    var info = await _signInManager.GetExternalLoginInfoAsync();
```

Comment:
```suggestion
    var externalLogin = await _signInManager.GetExternalLoginInfoAsync();
```

Reply:
@Tratcher  ... This is template driven, right? ...

https://github.com/dotnet/aspnetcore/blob/main/src/Identity/UI/src/Areas/Identity/Pages/V4/Account/ExternalLogin.cshtml.cs#L137

U sure that we should change it?

Reply:
Yeah I'd match the scaffolded code over clarity

Reply:
Darn. Maybe we should change the template...

Reply:
Yeah that's fair, feel free to file an issue or PR for that

Reply:
Change it back to `externalLogin`?

If you know that's what you'll go with, I'd rather not come back to this given the workload 🏃😅.

Reply:
Don't worry about it for now, any change wouldn't happen until 6.0.

Reply:
Yeah this looks great to me,!

Reply:
I'll merge this when the build system comes back online.